dask/bytes/core.py

```diff
@@ -1,14 +1,45 @@
+from __future__ import annotations
+
 import copy
 import os
+from typing import Any, Literal, overload
 
 from fsspec.core import OpenFile, get_fs_token_paths
 from fsspec.utils import infer_compression, read_block
 
 from dask.base import tokenize
-from dask.delayed import delayed
+from dask.delayed import Delayed, delayed
 from dask.utils import is_integer, parse_bytes
 
 
+@overload
+def read_bytes(
+    urlpath: str | list[str],
+    delimiter: bytes | None = None,
+    not_zero: bool = False,
+    blocksize: str | int | None = "128 MiB",
+    sample: int | str | bool = "10 kiB",
+    compression: str | None = None,
+    include_path: Literal[True] = True,
+    **kwargs: Any,
+) -> tuple[bytes, list[list[Delayed]], list[str]]:
+    ...
+
+
+@overload
+def read_bytes(
+    urlpath: str | list[str],
+    delimiter: bytes | None = None,
+    not_zero: bool = False,
+    blocksize: str | int | None = "128 MiB",
+    sample: int | str | bool = "10 kiB",
+    compression: str | None = None,
+    include_path: Literal[False] = False,
+    **kwargs: Any,
+) -> tuple[bytes, list[list[Delayed]], list[str]]:
```

Comment:
Shouldn't this `overload` *not* return a `list[str]`?

Reply:
yep 🤦‍♂️. If I make the change to `tuple[bytes, list[list[Delayed]]]` I get 
```
dask/bytes/core.py:16: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
```
from mypy. It looks like mypy doesn't see a difference between
```
include_path: Literal[False] = False
```
and
```
include_path: Literal[True] = True
```
because `include_path` is the only difference between the two overloads... still digging.

Reply:
OK looks like it had something to do with giving the overloads default arguments. I think it should be fixed :+1: